index.js

```diff
@@ -6,13 +6,19 @@ var through = require("through2");
 // # Filter
 // Create a through stream that only passes data that passes the given test
 // function
-module.exports = function(test, opt) {
+var filter = module.exports = function(test, opt) {
 	return through(opt || {}, function(data, _, cb) {
 		if (test(data)) cb(null, data);
 		else cb(null);
 	});
 };
 
+module.exports.obj = function(test, opt) {
+	opt = opt || {}
+	opt.objectMode = true
```

Comment:
Let's do this like [`through2` itself does it](https://github.com/rvagg/through2/blob/master/through2.js#L88): use `xtend` to not mutate the passed argument and set `highWaterMark`.